lntest/itest/lnd_onchain_test.go

```diff
@@ -379,3 +382,237 @@ func testAnchorReservedValue(net *lntest.NetworkHarness, t *harnessTest) {
+	)
+	if err != nil {
+		t.Fatalf("unable to get transaction: %v", err)
+	}
+	_, err = net.Miner.GenerateAndSubmitBlock(
+		[]*btcutil.Tx{commitTxn}, -1, emptyTime,
+	)
+	if err != nil {
+		t.Fatalf("unable to generate block: %v", err)
+	}
+
+	// With the anchor output located, and the main commitment mined we'll
+	// instruct the wallet to send all coins in the wallet to a new address
+	// (to the miner), including unconfirmed change.
+	minerAddr, err := net.Miner.NewAddress()
+	if err != nil {
+		t.Fatalf("unable to create new miner addr: %v", err)
+	}
+	sweepReq := &lnrpc.SendCoinsRequest{
+		Addr:             minerAddr.String(),
+		SendAll:          true,
+		MinConfs:         0,
+		SpendUnconfirmed: true,
+	}
+	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+	sweepAllResp, err := alice.SendCoins(ctxt, sweepReq)
+	if err != nil {
+		t.Fatalf("unable to sweep coins: %v", err)
+	}
+
+	// Next, we'll shutdown Alice, and allow 16 blocks to pass so that the
+	// anchor output can be swept by anyone. Rather than use the normal API
+	// call, we'll generate a series of _empty_ blocks here.
+	aliceRestart, err := net.SuspendNode(alice)
+	if err != nil {
+		t.Fatalf("unable to shutdown alice: %v", err)
+	}
+	const anchorCsv = 16
+	for i := 0; i < anchorCsv; i++ {
+		_, err := net.Miner.GenerateAndSubmitBlock(nil, -1, emptyTime)
+		if err != nil {
+			t.Fatalf("unable to generate block: %v", err)
+		}
+	}
+
+	// Before we sweep the anchor, we'll restart Alice.
+	//
+	// TODO(roasbeef): should do this after we sweep?
+	if err := aliceRestart(); err != nil {
+		t.Fatalf("unable to restart alice: %v", err)
+	}
+
+	// Now that the channel has been closed, and Alice has an unconfirmed
+	// transaction spending the output produced by her anchor sweep, we'll
+	// mine a transaction that double spends the output.
+	thirdPartyAnchorSweep := genAnchorSweep(t, net, aliceAnchor, anchorCsv)
+	_, err = net.Miner.GenerateAndSubmitBlock(
+		[]*btcutil.Tx{thirdPartyAnchorSweep}, -1, emptyTime,
+	)
+	if err != nil {
+		t.Fatalf("unable to generate block: %v", err)
+	}
+
+	// At this point, we should no longer find Alice's transaction that
+	// tried to sweep the anchor in her wallet.
+	assertTransactionNotInWallet(t.t, alice, aliceAnchor.SweepTx.TxHash())
+
+	// In addition, the transaction she sent to sweep all her coins to the
+	// miner also should no longer be found.
+	sweepAllTxID, _ := chainhash.NewHashFromStr(sweepAllResp.Txid)
+	assertTransactionNotInWallet(t.t, alice, *sweepAllTxID)
+
+	// The anchor should now show as being "lost", while the force close
+	// response is still present.
+	assertAnchorOutputLost(t, alice, chanPoint)
+
+	// At this point Alice's CSV output should already be fully spent and
+	// the channel marked as being resolved. We mine a block first, as so
+	// far we've been generating custom blocks this whole time..
+	commitSweepOp := wire.OutPoint{
+		Hash:  *forceCloseTxID,
+		Index: 1,
+	}
+	t.Log("writing for in input")
```

Comment:
huh?

Reply:
Stray debug log ;) -- will remove in a follow up fix up commit